src/lib.rs

```diff
@@ -234,6 +234,16 @@ impl Url {
     }
 
     /// Parse a string as an URL, with this URL as the base URL.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use url::Url;
+    ///
+    /// let url = Url::parse("https://example.net").unwrap();
+    /// let url = url.join("foo").unwrap();
+    /// assert_eq!(url.as_str(), "https://example.net/foo");
```

Comment:
Please change this to showcase trailing slashes: `https://example.net/foo` joined with `bar` is `https://example.net/bar`, but `https://example.net/foo/` joined with `bar` is `https://example.net/foo/bar`

r=me with this.